templates/default/fulldoc/html/frames.erb

```diff
@@ -5,10 +5,14 @@
 	<title><%= options.title %></title>
 </head>
 <script type="text/javascript">
-  var match = decodeURIComponent(window.location.hash).match(/^#!(.+)/);
-  var name = match ? match[1] : '<%= url_for_main %>';
-  name = name.replace(/^((\w*):)?[\/\\]*/gm, '').trim();
-  window.top.location.replace(name)
+try {
+    var match = decodeURIComponent(window.location.hash).match(/^#!(.+)/);
+    var name = match ? match[1] : '<%= url_for_main %>';
+    var url = new URL(name, location.href);
+    window.top.location.replace(url.origin === location.origin ? name : '<%= url_for_main %>');
```

Comment:
at this point you might want to assign `'<%= url_for_main %>'` to a variable.

Reply:
I don't think its needed for that. While it is commonly referenced, it's not stored in the browser memory until it is needed. I think this, while a bit uglier, is more browser-efficient.

Reply:
It's less efficient on the generation side since we're unnecessarily calling `url_for_main` 2 extra times. Also from a DRY perspective, it we should only have the one variable.

Reply:
Okay, would you like to me to update the code, or will you?